Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/compute_normal.h

```diff
@@ -49,7 +49,9 @@ namespace internal {
   {
     typename GT::FT norm = CGAL::approximate_sqrt(
         traits.compute_squared_length_3_object()(v));
-    v = traits.construct_divided_vector_3_object()(v, norm);
+    //If the vector is small enough, approx_sqrt might return 0, and then we get nan values. 
+    //To avoid that, we check the resulted norm. If it is 0, we don't normalize.
+    v = traits.construct_divided_vector_3_object()(v, norm == 0 ? 1 : norm);
```

Comment:
why don't you just use a `if(norm != 0)` ?
It avoids doin another construction

Reply:
Indeed.